File: app/src/main/java/com/karkoszka/cookingtime/common/CTColor.java
package com.karkoszka.cookingtime.common;

public class CTColor {
	private final int color;
	public int getColor() {
		return color;
	}

	public String getName() {
		return name;
	}

	private final String name;
	private final int backGroundColor;
	
	public CTColor(int color, String name, int backGroundColor) {
		this.color = color;
		this.name = name;
		this.backGroundColor = backGroundColor;
	}

	public CTColor(int color, String name) {
		this.color = color;
		this.name = name;
		this.backGroundColor = 0;
	}

	public int getBackGroundColor() {
		return backGroundColor;
	}
}

File: app/src/main/java/com/karkoszka/cookingtime/common/CTColor.kt
package com.karkoszka.cookingtime.common

class CTColor(val color: Int) {

    companion object {
        fun createArray(array: IntArray) : Array<CTColor?> {
            val values = arrayListOf<CTColor>()
            for (i in array) {
                values.add(CTColor(i))
            }
            return values.toTypedArray()
        }
    }
}